### PR TITLE
finish frontend_skeleton

### PR DESCRIPTION
**Pull request by siuwafong**
Completed ticket #15. Added some basic functionality, such as filtering list of friends in the search bar, typing messages to friends, and viewing messages from different friends.

### CODE REVIEW

**Comment by bonnieli** [comment]
Hi @siuwafong looks like your server/.gitignore file is not working, what you might want to do is change just `node_modules` to `node_modules/` and see if that actually ignores the whole directory, this pull request reflects 2098, and you're pushing up thousands of files onto Github which won't be sustainable since each user account has a max limit

**Comment by bonnieli** [comment]
You might also find this guide: https://www.freecodecamp.org/news/practical-git-and-git-workflows/#how-to-bring-our-local-up-to-date helpful in resolving the conflicting files you have

File: client/src/components/User.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const User = ({ username, profileImg, status, recentMsg, unreadMsgs }) => {
+    return (
+        <div className="userContainer">
+            <div className="userPicStatus">
+                <img className="userPic" src={profileImg} alt="userPic"/>
+                <div className={`statusDot ${status === "Online" ? "statusAvailable" : "statusAway"}` }></div> 
+            </div>
+            <div className="userNameMsg">
+                <p className="userName"> {username} </p>
```
Comment (sina-jamshidi) [inline client/src/components/User.js:12]: try to use [Typography](https://material-ui.com/components/typography/) when you need text

File: client/src/pages/Chat.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react"
+import User from '../components/User'
+import TextField from '@material-ui/core/TextField';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import SearchIcon from '@material-ui/icons/Search'
+import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
+import InsertEmoticonIcon from '@material-ui/icons/InsertEmoticon';
+import FileCopyIcon from '@material-ui/icons/FileCopy';
+import { MOCK_PROFILE, MOCK_FRIENDS, MOCK_MSGS } from '../utils/MockData'
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:9]: nice solution for now!

File: client/src/pages/Chat.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react"
+import User from '../components/User'
+import TextField from '@material-ui/core/TextField';
+import InputAdornment from '@material-ui/core/InputAdornment';
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:4]: To save space you can import from material UI by destructuring:
```javascript
import {TextField, InputAdornment} from '@material-ui/core`;
```

File: client/src/pages/Chat.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        setFriends(MOCK_FRIENDS.filter(friend => friend.username.includes(e.target.value)))
+    }
+
+    const toggleMsgInput = e => {
+        setNewMsg(e.target.value)
+    }
+
+    const toggleMsgSubmit = e => {
+        e.preventDefault()
+        console.log(newMsg)
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:48]: we can remove console.logs from the finished code

File: client/src/pages/Chat.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                    <div className="currentUserPicStatus">
+                        <img className="userPic" src={currentUser.img} alt="userPic"/>
+                        <div className={`statusDot ${currentUser.status === "Online" ? "statusAvailable" : "statusAway"} `}></div> 
+                    </div>
+                    <div className="userNameMsg">
+                        <p className="userName"> {currentUser.username} </p> 
+                    </div>
+                    <MoreHorizIcon />
+                </div>
+                <h3>Chats</h3>
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:74]: These could be `<Typography variant="h3">`

File: client/src/pages/Chat.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                <TextField
+                    className="chatSearchBar"
+                    placeholder="Search"
+                    variant="outlined"
+                    onChange={(e) => toggleSearch(e)}
+                    InputProps={{
+                        startAdornment: (
+                            <InputAdornment position="start">
+                                <SearchIcon />
+                            </InputAdornment>
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:84]: 👍

File: client/src/pages/Chat.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                        ),
+                    }}
+                />
+                <div className="usersContainer">
+                {friends.map(user => (
+                    <div 
+                        onClick={() => selectChat(user.id)}
+                        className={selectedChat !== null ? (selectedChat.id === user.id && `selectedChat`) : ""}
+                    >
```
Comment (sina-jamshidi) [inline client/src/pages/Chat.js:93]: should have a unique `key` prop here instead of in the `User`.

**Review by sina-jamshidi — COMMENTED** [review]
Suggested some changes.
Two points: 
- `gitignore` still needs to be fixed. You may need to also clear your git cache
- When there's a frontend PR, it's good practice to post screenshots and video captures that show the final product

**Comment by siuwafong** [comment]
> Hi @siuwafong looks like your server/.gitignore file is not working, what you might want to do is change just `node_modules` to `node_modules/` and see if that actually ignores the whole directory, this pull request reflects 2098, and you're pushing up thousands of files onto Github which won't be sustainable since each user account has a max limit

Sorry I forgot to make a .gitignore file for the server folder. I just made it right now.

**Comment by siuwafong** [comment]
> Suggested some changes.
> Two points:
> 
> * `gitignore` still needs to be fixed. You may need to also clear your git cache
> * When there's a frontend PR, it's good practice to post screenshots and video captures that show the final product

OK Sina. I'll make your suggested changes and I will post screenshots and video captures when I push my changes.

**Comment by siuwafong** [comment]
OK @bonnieli I read through the link you sent me and I typed ```git rm --cached -r server/node_modules```. I don't see the node_modules folder in the repo anymore. Please let me know if that still hasn't removed the ```node_modules``` folder.

File: client/src/components/ChatMsg.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                <div className="sentMsgBubble">{content} </div>
+                : 
+                <img src={content} alt="content" className="msgPic"/>}
+            </div>
+        )
+    }
+    
+    else {
+        return (
+            <div className="receivedMsg">
```
[the remark this answers is not in the document]
Comment (siuwafong) [reply client/src/components/ChatMsg.js:23]: OK I will replace all my divs with Grid

File: client/src/pages/Chat.js (hunk 2 above)
Comment (siuwafong) [reply client/src/pages/Chat.js:9]: thanks!

File: client/src/components/ChatMsg.css [hunk 9]
```diff
@@ -0,0 +1,40 @@
+.receivedMsgPic {
```
[the remark this answers is not in the document]
Comment (siuwafong) [reply client/src/components/ChatMsg.css:1]: I'm not sure about the difference between the two. I think in a previous pull request Bonnie said one of them involved HOCs and I shouldn't use it. I used "useStyles".

**Comment by siuwafong** [comment]
https://www.loom.com/share/5b6134b9d80e452d8c5876501c969404

File: client/src/pages/Chat.css [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+.hideCurrent {
+    display: hidden;
+    visibility: none;
+}
+
+.chatMsg {
+    width: 80%;
+    text-align: center;
+    /* position: absolute; */
```
Comment (bonnieli) [inline client/src/pages/Chat.css:93]: remove

**Comment by bonnieli** [comment]
> OK @bonnieli I read through the link you sent me and I typed `git rm --cached -r server/node_modules`. I don't see the `node_modules` folder in the repo anymore. Please let me know if that still hasn't removed the `node_modules` folder.

Super! this PR looks accurate now, 18 files changed :) 

you did a lot more than expected for ticket #15. in the future you can expect to do smaller and smaller sized PRs, this will help you manage each ticket better :)

**Review by bonnieli — APPROVED** [review]
Merge conflicts, please resolve them and this has been approved so after resolving conflicts you're free to merge this in! :D

**Pull request merged**